### ISSUE TITLE
Support more pretrained models for Transformer Encoder

### ISSUE THREAD

**Issue by festeh** [jina-ai/jina#691]
**Describe the feature**
 After this proposal users will be able to use all of the models from this list (https://huggingface.co/models) while using `BaseTransformerEncoder`. Not all of these models are intended for for information retrievial tasks, but some may be useful. Note that with the current implementation users are already having issues, see #662.

**Your proposal**
Use this [class](https://huggingface.co/transformers/model_doc/auto.html#automodelforpretraining) from transformers library.
The code is already implemented in my fork [here](https://github.com/festeh/jina/blob/feat-support-more-transformers/jina/executors/encoders/nlp/transformer.py), if you're interested, I can polish it a bit and make a PR.

Pros:
* Users will be able to load all available pre-trained models
* The number of lines of code is reduced, removed hardcoded imports

Cons:
* Hard-coding will now occur in other place. For each model one needs to decide how to get a sequence embedding from the embeddings of its tokens. I propose a simple heuristic: if a model is BERT-ish, take embedding of a first token, otherwise (XLNet, GPT-2 and other language models), take embedding of a last token. There's no guarantee that it would work for all models
* Did not test for all possible models, just considered existing corner cases (need to add a padding token for language model)

**Comment by JoanFM** [comment]
hey @festeh,

As you say, this has caught already the attention of more people. You are welcome to create a Pull Request where we can see the details and discuss about this.
